Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift

```diff
@@ -72,239 +73,206 @@ public final class SwiftPackageManagerGraphGenerator: SwiftPackageManagerGraphGe
+            infoPlist: .default,
+            sources: target.mapSources(path: path),
+            resources: target.mapResources(path: path),
+            dependencies: try target.mapDependencies(packageName: packageName, packageInfo: packageInfo, productToPackage: productToPackage),
+            settings: try target.mapSettings()
+        )
+    }
+}
 
-                var cHeaderSearchPaths: [String] = []
-                var cxxHeaderSearchPaths: [String] = []
-                var cDefines: [String: String] = [:]
-                var cxxDefines: [String: String] = [:]
-                var swiftDefines: [String: String] = [:]
-                var cFlags: [String] = []
-                var cxxFlags: [String] = []
-                var swiftFlags: [String] = []
-
-                try target.settings.forEach { setting in
-                    let platforms = try setting.condition?.platforms()
-                    switch (setting.tool, setting.name) {
-                    case (.c, .headerSearchPath):
-                        cHeaderSearchPaths.append(setting.value[0])
-                    case (.c, .define):
-                        let (name, value) = setting.extractDefine
-                        cDefines[name] = value
-                    case (.c, .unsafeFlags):
-                        cFlags.append(contentsOf: setting.value)
-
-                    case (.cxx, .define):
-                        let (name, value) = setting.extractDefine
-                        cxxDefines[name] = value
-                    case (.cxx, .headerSearchPath):
-                        cxxHeaderSearchPaths.append(setting.value[0])
-                    case (.cxx, .unsafeFlags):
-                        cxxFlags.append(contentsOf: setting.value)
-
-                    case (.swift, .define):
-                        let (name, value) = setting.extractDefine
-                        swiftDefines[name] = value
-                    case (.swift, .unsafeFlags):
-                        swiftFlags.append(contentsOf: setting.value)
-
-                    case (.linker, .linkedFramework):
-                        dependencies.append(.linkedFramework(name: setting.value[0], platforms: platforms))
-                    case (.linker, .linkedLibrary):
-                        dependencies.append(.linkedLibrary(name: setting.value[0], platforms: platforms))
-                    case (.c, .linkedFramework), (.c, .linkedLibrary), (.cxx, .linkedFramework), (.cxx, .linkedLibrary),
-                         (.swift, .headerSearchPath), (.swift, .linkedFramework), (.swift, .linkedLibrary),
-                         (.linker, .headerSearchPath), (.linker, .define), (.linker, .unsafeFlags):
-                        throw SwiftPackageManagerGraphGeneratorError.unsupportedSetting(setting.tool, setting.name)
+extension PackageInfo.Target {
+    func mapPlatform() -> ProjectDescription.Platform {
+        // TODO: Should this be configured in `Dependencies.swift`?
+        return .iOS
```

Comment:
How would you define the platform? should it be defined in the Dependencies.swift file? 🤔

Reply:
You can define platforms in `Dependencies.swift`. Check [here](https://github.com/tuist/tuist/blob/c828fe563d66662902d8f7c3838901f258acb347/Sources/ProjectDescription/Dependencies/Dependencies.swift#L12).

I use them for `Carthage` installation [here](https://github.com/tuist/tuist/blob/c828fe563d66662902d8f7c3838901f258acb347/Sources/TuistDependencies/Carthage/CarthageInteractor.swift#L83).

Reply:
Nice, thanks 🙌 
The problem is that here we can specify only one platform 😟

Reply:
yeah, `Project` does not support multiple platforms, unfortunately. This is a bit of an issue 😁 I think for the time being we can default to iOS or _some_ platform if iOS is not available. This is obviously a temporary measure and before we mark `Dependencies.swift` usable, we should finish off [this](https://github.com/tuist/tuist/pull/434) PR that will enable us to specify multiple platforms for a project.

I'd also add a compiler warning, so it's something we don't forget about as it's important to do this right - it's just out of scope of this PR.